### PR TITLE
allow sending of TLP and RTO probe packets when congestion limited

### PR DESCRIPTION
**Pull request by marten-seemann**
We already have text allowing to send TLP and RTO packets when congestion limited in the respective TLP and RTO sections.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -885,7 +885,9 @@ specified in bytes due to finer control and the ease of appropriate byte
 counting {{?RFC3465}}.
 
 QUIC hosts MUST NOT send packets if they would increase bytes_in_flight
-(defined in {{vars-of-interest}}) beyond the available congestion window.
+(defined in {{vars-of-interest}}) beyond the available congestion window, unless
+the packet is a probe packet sent after the TLP or RTO alarm fires, as described
+in {{tlp}} and {{rto}}.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]: Now it looks like this is a definitive list, but we have an exemption for packets that only carry ACK frames as well.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: ACK only packets don't increase bytes_in_flight, but that does remind me that handshake packets also may increase bytes_in_flight above the congestion window.

Comment (marten-seemann) [reply draft-ietf-quic-recovery.md]: Handshake packets are not congestion limited? Where does it say so?

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: Sorry, I meant that similarly to TLP or RTO packets, handshake packets count towards bytes in flight, but can be sent even if bytes in flight exceeds congestion window.

Comment (marten-seemann) [reply draft-ietf-quic-recovery.md]: Can you point me to the section where this is specified? I can't find it.

Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I had this in mind: "When the alarm fires, the sender MUST retransmit all unacknowledged handshake data. On each consecutive firing of the handshake alarm, the sender SHOULD double the handshake timeout and set an alarm for this period."

**Pull request merged**